system/jlib/jmetrics.hpp

```diff
@@ -262,4 +273,50 @@ std::shared_ptr<CustomMetric<T>> createCustomMetricAndAddToReporter(const char *
     return pMetric;
 }
 
+
+class jlib_decl ScopedGaugeUpdater
+{
+public:
+    explicit ScopedGaugeUpdater(const std::shared_ptr<GaugeMetric> &_pGauge, int64_t _amount=1) :
+        gauge{*_pGauge},
+        amount{_amount}
+        {
+            gauge.inc(amount);
+        }
+    ScopedGaugeUpdater(const ScopedGaugeUpdater&) = delete;
+    ScopedGaugeUpdater(ScopedGaugeUpdater&) = delete;
+    ScopedGaugeUpdater& operator=(const ScopedGaugeUpdater&) = delete;
+    ScopedGaugeUpdater& operator=(ScopedGaugeUpdater&) = delete;
+
+    ~ScopedGaugeUpdater()
+    {
+        gauge.dec(amount);
+    }
+
+protected:
+    GaugeMetric &gauge;
+    int64_t amount;
+};
+
+
+class jlib_decl AutoGaugeDecrement
```

Comment:
it's still a scoped utiltiy class, to be consistent with other one (ScopedGaugeUpdater), I think this should be called ScopedGagueDecrement